src/js/ripple/seed.js

```diff
@@ -57,104 +62,105 @@ Seed.prototype.parse_passphrase = function(j) {
+        lodash.isEqual(result.version, Base.VER_ED25519_SEED)) {
+      this._version = result.version;
+      this._value = sjcl.bn.fromBits(sjcl.codec.bytes.toBits(result.bytes));
+  }
+
+  return this;
+};
+
 Seed.prototype.to_json = function() {
   if (!(this.is_valid())) {
     return NaN;
   }
 
-  var output = Base.encode_check(Base.VER_FAMILY_SEED, this.to_bytes());
+  var version = this._version; // Base.VER_FAMILY_SEED;
+  var output = Base.encode_check(version, this.to_bytes());
 
   return output;
 };
 
-function append_int(a, i) {
-  return [].concat(a, i >> 24, (i >> 16) & 0xff, (i >> 8) & 0xff, i & 0xff);
-}
-
-function firstHalfOfSHA512(bytes) {
-  return sjcl.bitArray.bitSlice(
-    sjcl.hash.sha512.hash(sjcl.codec.bytes.toBits(bytes)),
-    0, 256
-  );
-}
-
-// Removed a `*` so this JSDoc-ish syntax is ignored.
-// This will soon all change anyway.
-/*
-* @param account
-*        {undefined}                 take first, default, KeyPair
-*
-*        {Number}                    specifies the account number of the KeyPair
-*                                    desired.
-*
-*        {Uint160} (from_json able), specifies the address matching the KeyPair
-*                                    that is desired.
+Seed.prototype.get_key = function(opts) {
+  if (opts !== undefined && typeof opts !== 'object') {
+    throw new Error('get_key options not supported: ' + opts);
+  }
+  return this._get_key(opts);
+};
+
+/**
+* @return {KeyPair} - the root key-pair, as used by validators.
+*/
+Seed.prototype.get_root_key = function() {
+  return this._get_key({root: true});
+};
+
+/**
+* @param {Object} [options] -
 *
-* @param maxLoops (optional)
-*        {Number}                    specifies the amount of attempts taken
-*                                    to generate a matching KeyPair
+* @param {Number} [options.accountIndex=0] - the account number to generate
 *
+* @param {Boolean} [options.root=false] - generate root key-pair,
+*                                         as used by validators.
+* @return {KeyPair} -
 */
-Seed.prototype.get_key = function(account, maxLoops) {
-  var account_number = 0, address;
-  var max_loops = maxLoops || 1;
+Seed.prototype._get_key = function(options) {
+  var opts = options || {};
+
+  var root = opts.root;
 
   if (!this.is_valid()) {
     throw new Error('Cannot generate keys from invalid seed!');
   }
-  if (account) {
-    if (typeof account === 'number') {
-      account_number = account;
-      max_loops = account_number + 1;
-    } else {
-      address = UInt160.from_json(account);
-    }
-  }
 
-  var private_gen, public_gen;
+  var privateGen, publicGen;
   var curve = this._curve;
   var i = 0;
 
   do {
-    private_gen = sjcl.bn.fromBits(
-            firstHalfOfSHA512(append_int(this.to_bytes(), i)));
-    i++;
-  } while (!curve.r.greaterEquals(private_gen));
+    // We hash the seed to extend from 128 bits to 256, looping until we are
```

Comment:
I talked to a math whiz, who said he had the same curiosity tickled, and he said that if you brute force intelligently, it takes only the square root of the bits used for the private keys. So 2**256 only needs 2**128 operations. I don't really understand it, but I do understand it's what rippled does to get the private key from a seed, and can't be changed easily.